### PR TITLE
feat: Import and export counterfactual safes

### LINKED ISSUE #3959: Counterfactual Safes should be included in the export/import
Atm, exporting your data as a JSON file doesn't include counterfactual accounts. They should be there as well.

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #3959 

## How this PR fixes it

- Adds the undeployedSafesSlice to the import and export page in settings

## How to test it

1. Create a counterfactual safe
2. Go to the Settings > Data
3. Export your Data
4. Observe it displays that an undeployed safe will be exported
5. Open a new Browser or clear your local storage
6. Go to the Settings > Data page
7. Import the file
8. Observe it displays that an undeployed safe will be imported
9. Observe the safe appearing in the sidebar
10. Observe no errors when opening the safe

## Screenshots
Screenshot 2024-07-22 at 11 11 52

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/settings/DataManagement/FileListCard.tsx [hunk 1]
```diff
@@ -129,6 +132,18 @@ const getItems = ({
     items.push(safeAppsPreview)
   }
 
+  if (undeployedSafes) {
+    const undeployedSafesPreview: ListItemTextProps = {
+      primary: (
+        <>
+          <b>Undeployed Safes</b> {Object.keys(undeployedSafes).length}
```
Comment (katspaugh) [inline src/components/settings/DataManagement/FileListCard.tsx:139]:
```suggestion
          <b>Not activated Safe Accounts</b> {Object.keys(undeployedSafes).length}
```

**Review by compojoom — APPROVED** [review]
beautiful! I checked it and my counterfactual safes were properly exported and imported.

**Comment by francovenica** [comment]
Importing exporting works fine.

Only one issue: the counter shows that I'll be exporting 1 not-activated safe account, when I have 2 in memory (since I had 2 
different owners creating each one)

image

**Comment by usame-algan** [comment]
> Only one issue: the counter shows that I'll be exporting 1 not-activated safe account, when I have 2 in memory (since I had 2 different owners creating each one)

Good catch! I fixed the calculation and it should now display the correct number of undeployed safes across chains and owners.

**Comment by francovenica** [comment]
Issue fixed. LGTM

**Pull request merged**